substrate/frame/utility/src/lib.rs

```diff
@@ -454,6 +458,53 @@ pub mod pallet {
 			let res = call.dispatch_bypass_filter(frame_system::RawOrigin::Root.into());
 			res.map(|_| ()).map_err(|e| e.error)
 		}
+
+		#[pallet::call_index(6)]
+		#[pallet::weight(10_000)]
+		pub fn if_else(
+			origin: OriginFor<T>,
+			main: <T as Config>::RuntimeCall,
+			fallback: <T as Config>::RuntimeCall,
+		) -> DispatchResultWithPostInfo {
+			// Do not allow the `None` origin.
+			if ensure_none(origin.clone()).is_ok() {
+				return Err(BadOrigin.into())
+			}
+	
+			let is_root = ensure_root(origin.clone()).is_ok();
+			let calls: Vec<<T as Config>::RuntimeCall> = vec![main.clone(), fallback.clone()];
+	
+			// Track the weights
+			let mut weight = Weight::zero();
+			for (index, call) in calls.into_iter().enumerate() {
```

Comment:
I dont quite understand what is going on.  

What we want is an if-else, there should be no vector or iteration needed.

Reply:
Thanks for the review!

Made adjustments